linera-base/src/crypto/secp256k1/evm.rs

```diff
@@ -65,16 +65,15 @@ pub struct EvmKeyPair {
 
 /// A secp256k1 signature.
 #[derive(Eq, PartialEq, Copy, Clone)]
-pub struct EvmSignature(pub(crate) Signature);
+pub struct EvmSignature(pub Signature);
 
 impl FromStr for EvmSignature {
     type Err = CryptoError;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         // If the string starts with "0x", we remove it before decoding.
         let bytes = hex::decode(s.strip_prefix("0x").unwrap_or(s))?;
-        let sig = Signature::from_erc2098(&bytes);
```

Comment:
Or would it be possible to do the same thing as [EIP-2](https://eips.ethereum.org/EIPS/eip-2), and generate only signatures with the first s-value bit 0?

Reply:
We're relying on alloy to handle all the possible cases. The problem here was that the input bytes were interpreted incorrectly - i.e. they assumed the format was different, that the parity bit (recovery) was encoded into `s` while it's appended `[s, r, v]` (32bytes, 32bytes, 1byte).